co2eq/purchase/index.js

```diff
@@ -163,24 +197,14 @@ export function carbonEmissions(activity) {
       break;
 
     case ACTIVITY_TYPE_PURCHASE: {
-      const { purchaseType } = activity;
-      const entry = getEntryByKey(purchaseType);
-      if (!entry) {
-        throw new Error(`Unknown purchaseType: ${purchaseType}`);
-      }
-      if (!entry.intensityKilograms) {
-        throw new Error(`Missing carbon intensity for purchaseType: ${purchaseType}`);
-      }
-
-      const { unit, amount } = extractComptabileUnitAndAmount(activity, entry);
-      if (unit == null || amount == null || !Number.isFinite(amount)) {
-        throw new Error(`Invalid unit ${unit} or amount ${amount} for purchaseType ${purchaseType}. Expected ${entry.unit}`);
-      }
+      const { lineItems } = activity;
 
-      if (entry.unit !== unit) {
-        throw new Error(`Invalid unit ${unit} given for purchaseType ${purchaseType}. Expected ${entry.unit}`);
+      // First check if lineItems contains and calculate total of all line items
+      if (lineItems && lineItems.length) {
+        // TODO(df): What to do on a single line error? Abort all? Skip item?
+        footprint = lineItems.map(l => carbonEmissionOfLineItem(l)).reduce((a, b) => a + b, 0);
       }
-      footprint = entry.intensityKilograms * amount;
+      if (!footprint) throw new Error(`Could not calculate carbonIntensity of purchase activity with lineItems: ${JSON.stringify(lineItems)}`)
```

Comment:
no need to stringify -- the error will have the `activity` as contextMessage.
Futhermore, you don't even need to throw the error (I think) as if `footprint` is 0 or null an error will be thrown outside of this.

Reply:
Hm I just checked, if returning `0` or `null` as `footprint` would actually go through and the activity is stored:

![Simulator Screen Shot - iPhone SE - 2020-03-30 at 10 23 27](https://user-images.githubusercontent.com/5886935/77890947-83b80400-7270-11ea-8748-c47a7fe98bbb.png)

Reply:
Ah you're right indeed.

Reply:
Hm tricky one: An error is actually thrown in `carbonmodelselector`, but immediately caught and then a `carbonModelNextRunDate` is set (for electricity for example). 

Should we remove as proposed the validation check and handle it in the internal repo's PR?

Reply:
yep I would remove it from here and handle it in the internal repo